### PR TITLE
[Privacy] Send client side metrics only when consent given

### PR DESCRIPTION
**Pull request by vguerci**

### CODE REVIEW

File: CriteoPublisherSdk/Sources/Network/CR_CdbResponse.h [hunk 1]
```diff
@@ -29,6 +29,7 @@ NS_ASSUME_NONNULL_BEGIN
 
 @property(copy, nonatomic, nullable) NSArray<CR_CdbBid *> *cdbBids;
 @property(assign, nonatomic) NSUInteger timeToNextCall;
+@property(assign, nonatomic, readonly) BOOL consentGiven;
```
Comment (TurpIF) [inline CriteoPublisherSdk/Sources/Network/CR_CdbResponse.h:32]: You added the consent flag but you didn't set it in the `.m` file.

**Review by TurpIF — COMMENTED** [review]
LGTM for now.

And oups I didn't saw this was a draft, but I reviewed the code anyway :)

File: CriteoPublisherSdk/Sources/Network/CR_CdbResponse.h (hunk 1 above)
Comment (vguerci) [reply CriteoPublisherSdk/Sources/Network/CR_CdbResponse.h]: > LGTM for now.
> 
> And oups I didn't saw this was a draft, but I reviewed the code anyway :)

Yup sorry forgot to push this as draft directly, this is mostly to run all tests ;p

**Review by TurpIF — DISMISSED** [review]

**Review by TurpIF — APPROVED** [review]

**Comment by vguerci** [comment]
Thanks!

**Pull request merged**